internal/util/inventoryconverter.go

```diff
@@ -0,0 +1,138 @@
+package inventoryconverter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bmc-toolbox/common"
+	fleetdb "github.com/metal-toolbox/fleetdb/pkg/api/v1"
+	rivets "github.com/metal-toolbox/rivets/types"
+)
+
+type InventoryConverter struct {
+	componentSlug map[string]*fleetdb.ServerComponentType
+}
+
+func NewInventoryConverter(componentSlug map[string]*fleetdb.ServerComponentType) *InventoryConverter {
+	return &InventoryConverter{
+		componentSlug: componentSlug,
+	}
+}
+
+func (ic *InventoryConverter) ToRivetsServer(serverID, facility string, device *common.Device, biosCfg map[string]string) (*rivets.Server, error) {
+	components, err := ic.getComponentSlice(device)
+	if err != nil {
+		return nil, err
+	}
+
+	return &rivets.Server{
+		BIOSCfg:  biosCfg,
+		ID:       serverID,
+		Facility: facility,
+		// Name:
+		// BMCAddress:
+		// BMCUser:
+		// BMCPassword: device.BMC.
```

Comment:
Hi Vince,
I am not sure do we want to fill in BMC credentials info?
If so, do we want to let Alloy sending those info to use? Or do we query the credentials inside CIS?

Reply:
For our purposes here skip the BMC details. The Name can be set as just the ID uuid as a string. In order for CIS to have any of that stuff Alloy would have to send it along, but those details came from FleetDB in the first place. Let's just avoid it.